### PR TITLE
Readme redo

### PR DESCRIPTION
**Pull request by GloreenS**
### Related links
Cards [#212](https://github.com/casper-network/governance/issues/212)

### Changes
Added test description and deploy command to deploy the session code on NCTL network.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,23 +1,197 @@
...
+## Build and Test the Session Code 
 
 ### Set up the Rust toolchain
 You need the Rust toolchain to develop smart contracts.
 ```bash
-$ make prepare
+make prepare
 ```
 
 ### Compile smart contracts
```
Comment (darthsiroftardis) [inline README.md]:
```suggestion
### Compile session code
```

File: README.md [hunk 2]
```diff
@@ -1,23 +1,197 @@
...
-Compile WASM files that will be used later.
+Compile WebAssembly (Wasm) files that will be used to deploy the session code.
```
Comment (darthsiroftardis) [inline README.md]:
```suggestion
Compile the session code to WebAssembly (Wasm)
```

File: README.md [hunk 3]
```diff
@@ -1,23 +1,197 @@
...
+Run the tests suite.
+```bash
+make test
+```
+
+## Deploy the Hello World Session Code
+
+You can deploy the Hello World session code on a local network using NCTL. For more information on how to run an NCTL network, see [Setting up an NCTL network](https://docs.casperlabs.io/dapp-dev-guide/building-dapps/setup-nctl/).
+
+This command provides a view into the faucet account details. The faucet is the default account created on the NCTL network.
```
Comment (darthsiroftardis) [inline README.md]:
```suggestion
This command provides a view into the faucet account details. The faucet is the special pre-funded Account created on the NCTL network.
```

**Review by darthsiroftardis — APPROVED** [review]
Optimistic approval pending nits

File: README.md [hunk 4]
```diff
@@ -1,23 +1,197 @@
-# Hello World Smart Contract
+# Hello World Session Code
 
-Simple hello world smart contract for Casper.
+This session code accepts a message string and stores it in the calling account under the `special_value` NamedKey.
 
-## Install
+**Usage**: This session code expects a runtime argument named `message` of type string.
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
**Usage**: This session code expects a runtime argument named `message` of the type `String`.
```

File: README.md [hunk 5]
```diff
@@ -1,23 +1,197 @@
...
+
+**Tests**: There are two tests available to test the Hello World session code. The `should_store_hello_world` test verifies the happy path, where a string *hello world* is saved under the `special_value` NamedKey. The `should_error_on_missing_runtime_arg` test verifies that an error is displayed when the runtime argument is missing.
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
**Tests**: There are two tests available to test the Hello World session code. The `should_store_hello_world` test verifies the successful path, where a string *hello world* is saved under the `special_value` NamedKey. The `should_error_on_missing_runtime_arg` test verifies that an error is displayed when the runtime argument is missing. 
```

File: README.md [hunk 6]
```diff
@@ -1,23 +1,197 @@
...
+# Hello World Session Code
 
-Simple hello world smart contract for Casper.
+This session code accepts a message string and stores it in the calling account under the `special_value` NamedKey.
 
-## Install
+**Usage**: This session code expects a runtime argument named `message` of type string.
+
+**Tests**: There are two tests available to test the Hello World session code. The `should_store_hello_world` test verifies the happy path, where a string *hello world* is saved under the `special_value` NamedKey. The `should_error_on_missing_runtime_arg` test verifies that an error is displayed when the runtime argument is missing. 
+The tests start by initializing the casper crates and creating a genesis account. Then the contract Wasm is loaded to the session_code object. The deploy_item object is created using the details like payment method, session code arguments, and account address. The deploy_item object is passed to the execute_request. Finally, the execution engine is invoked to process the execute_request.
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
The tests start by initializing the Casper crates and creating a genesis account. Then the contract Wasm is loaded to the session_code object. The deploy_item object is created using the details like payment method, session code arguments, and the account address. The deploy_item object is passed to the execute_request. Finally, the execution engine is invoked to process the execute_request. 
```

File: README.md [hunk 7]
```diff
@@ -1,23 +1,197 @@
...
+    }
+  }
+}
 ```
+
+</details>
+
+:::note
+
+Make a note of the NakedKey uref hash in the faucet account details.
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
Make a note of the NamedKey URef hash in the faucet account details.
```

File: README.md [hunk 8]
```diff
@@ -1,23 +1,197 @@
...
+
+:::
+
+### View the value stored in the NamedKey
+
+You can view the value stored in the NamedKey by using the uref hash assigned to the NamedKey. To do this, we will first find the state root hash of the network using the following command:
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
You can view the value stored in the NamedKey by using the URef hash assigned to the NamedKey. To do this, we will first find the state root hash of the network using the following command:
```

File: README.md [hunk 9]
```diff
@@ -1,23 +1,197 @@
...
+  "result": {
+    "api_version": "1.0.0",
+    "state_root_hash": "6097c728c1af6179a491b5c3d143c49b032c9d6bb69be553729cb9f1489c3833"
+  }
+}
+```
+
+</details>
+
+Then use the uref hash and the state root hash to view the value stored in the NamedKey.
```
Comment (ACStone-MTS) [inline README.md]:
```suggestion
Then use the URef hash and the state root hash to view the value stored in the NamedKey.
```

**Review by ACStone-MTS — APPROVED** [review]
A few minor fixes, but mostly looks good to me!

**Pull request merged**